knotx-core/knotx-repository/src/main/java/com/cognifide/knotx/repository/impl/HttpRepository.java

```diff
@@ -73,19 +70,20 @@ static HttpRepository of(String path, String domain, Integer port, Vertx vertx)
                 );
     }
 
-    private Observable<Buffer> reduceBuffers(HttpClientResponse response) {
+    private Observable<RepositoryResponse> processResponse(final HttpClientResponse response) {
         return Observable.just(Buffer.buffer())
                 .mergeWith(response.toObservable())
-                .reduce(Buffer::appendBuffer);
+                .reduce(Buffer::appendBuffer)
+                .flatMap(buffer -> toRepositoryResponse(buffer,response));
     }
 
-    private Observable<RepositoryResponse> toRepositoryResponse(Buffer buffer) {
+    private Observable<RepositoryResponse> toRepositoryResponse(Buffer buffer, final HttpClientResponse httpClientResponse) {
         Observable<RepositoryResponse> response;
         if (buffer.length() > 0) {
             response = RepositoryResponse.success(buffer.toString()).toObservable();
         } else {
-            LOGGER.error("Remote repository returned empty template for path `{}`", path);
-            response = RepositoryResponse.error("No Template found for path %s", path).toObservable();
+            LOGGER.info("Remote repository returned empty template for path `{}` with status code {}", path, httpClientResponse.statusCode());
+            response = RepositoryResponse.error(httpClientResponse.statusCode(), httpClientResponse.headers()).toObservable();
```

Comment:
As we discussed. I've created an issue #48 to discuss how to react on other status codes than 200/400/500